File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n=nums.length;
        for(int i=1;i<n;i++){
            for(int j=i;j>=1;j--){
                if(nums[j]>nums[j-1]){
                    int temp=nums[j];
                    nums[j]=nums[j-1];
                    nums[j-1]=temp;
                }
                else{
                    break;
                }
            }
        }
        return nums[n/2];
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        map<int,int>mp;
        int res;
        for(int num:nums){
            mp[num]++;
            if(mp[num]>=n/2 && mp[res]<mp[num]){
                res=num;
            }
        }
        return res;
    }
};